### PR TITLE
Added Loading state to submit Button

### PR DESCRIPTION
**Pull request by iddify**

### CODE REVIEW

**Comment by iddify** [comment]
don't forget to label as 'hacktoberfest-accepted' :)

File: .env.example [hunk 1]
```diff
@@ -1 +1 @@
-GATSBY_API_URL=url
+GATSBY_API_URL='https://quiet-woodland-08078.herokuapp.com/socket.io/'
```
Comment (jportella93) [inline .env.example:1]: This shouldn't be here

**Review by jportella93 — CHANGES_REQUESTED** [review]
Can you commit only the changes related to the loading state? all those formatting changes are unneeded here.

File: .env.example (hunk 1 above)
Comment (iddify) [reply .env.example]: resolved!

**Comment by iddify** [comment]
> Can you commit only the changes related to the loading state? all those formatting changes are unneeded here.

working on it

File: src/components/selectRoom.js [hunk 2]
```diff
@@ -1,34 +1,51 @@
...
+          name='userName'
+          htmlfor='userName'
+          label='User name'
+        >
+          <TextInput id='userName' name='userName' />
         </FormField>
-        <Box pad={{ top: 'xlarge' }} align="end">
-          <Button pad={{ top: 'large' }} type="submit" primary label="Submit" />
+        <Box pad={{ top: 'xlarge' }} align='end'>
+          <Button
```
Comment (jportella93) [inline src/components/selectRoom.js:42]: Unneeded formatting

File: src/components/selectRoom.js [hunk 3]
```diff
@@ -1,34 +1,51 @@
...
+            icon={isLoading ? <Spinner /> : null}
+            pad={{ top: 'large' }}
+            type='submit'
+            primary
```
Comment (jportella93) [inline src/components/selectRoom.js:46]: Unneeded formatting

File: src/components/socketConnection.js [hunk 4]
```diff
@@ -10,18 +10,17 @@ const SocketConnection = () => {
 
   const actions = {
     connectToRoom: (roomName, userName) => {
+      setRoomState({ ...roomState, isLoading: true })
       socket.current = socketIOClient(process.env.GATSBY_API_URL, {
-        query: `roomName=${roomName}&userName=${userName}`
+        query: `roomName=${roomName}&userName=${userName}`,
```
Comment (jportella93) [inline src/components/socketConnection.js:15]: Unneeded formatting

File: src/components/socketConnection.js [hunk 5]
```diff
@@ -38,26 +37,28 @@ const SocketConnection = () => {
...
 
   const derivedRoomState = {
     isAdmin: roomState.userType === 'admin',
     currentTeam,
-    isCurrentTeamTurn: roomState.room?.currentPlayingTeam?.id === currentTeam?.id,
-    isTeamLeader: currentTeam?.leader === roomState.userId
+    isCurrentTeamTurn:
+      roomState.room?.currentPlayingTeam?.id === currentTeam?.id,
+    isTeamLeader: currentTeam?.leader === roomState.userId,
   }
```
Comment (jportella93) [inline src/components/socketConnection.js:63]: Unneeded formatting

File: src/pages/index.js [hunk 6]
```diff
@@ -6,11 +6,11 @@ import SocketConnection from '../components/socketConnection'
...
   <Layout>
-    <SEO title="Home" />
+    <SEO title='Home' />
     <Box
       width={{ max: '500px', min: '50vw' }}
-      alignSelf="center"
-      alignContent="stretch"
+      alignSelf='center'
+      alignContent='stretch'
     >
```
Comment (jportella93) [inline src/pages/index.js:14]: Unneeded formatting

**Review by jportella93 — CHANGES_REQUESTED** [review]
Still some unneeded formatting

**Review by jportella93 — APPROVED** [review]
Now it's perfect, thanks! :) I'll merge and approve it to hacktoberfest

**Pull request merged**